blueflood-core/src/main/java/com/rackspacecloud/blueflood/outputs/handlers/RollupHandler.java

```diff
@@ -287,24 +305,109 @@ private Boolean repairMetrics (Locator locator, MetricData metricData, final lon
         return retValue;
     }
 
-    private List<Points.Point> repairRollupsOnRead(Locator locator, Granularity g, long from, long to) {
+    /**
+     * This method gets the points from the DB and then rolls them up according to the granularity.
+     *
+     * Breaks up the number of ranges into buckets based on ROLLUP_ON_READ_REPAIR_SIZE_PER_THREAD and executes
+     * the buckets in parallel.
+     *
+     * @param locator metric key within the DB
+     * @param g the granularity
+     * @param from the starting timestamp of the range (ms)
+     * @param to the ending timestamp of the range (ms)
+     *
+     * @return a list of rolled-up points
+     */
+    private List<Points.Point> repairRollupsOnRead(final Locator locator, Granularity g, long from, long to) {
+        Timer.Context c = timerRepairRollupsOnRead.time();
+
+        List<Points.Point> repairedPoints = new ArrayList<Points.Point>();
+        List<ListenableFuture<List<Points.Point>>> futures = new ArrayList<ListenableFuture<List<Points.Point>>>();
+
+        for( final Iterable<Range> ranges : divideRangesByGroup( g, from, to ) ) {
+            futures.add(
+
+                    createRepairPointsExecutor.submit( new Callable() {
+
+                        @Override
+                        public List<Points.Point> call() throws Exception {
+                            return createRepairPoints( ranges, locator );
+                        }
+                    } ) );
+        }
+
+        ListenableFuture<List<List<Points.Point>>> aggregateFuture = Futures.allAsList(futures);
+
+        try {
+            for( List<Points.Point> subList : aggregateFuture.get(rollupOnReadTimeout.getValue(), rollupOnReadTimeout.getUnit()) ) {
+
+                repairedPoints.addAll( subList );
+            }
+        } catch (Exception e) {
+            log.warn(String.format("Exception encountered while doing rollups on read, incomplete rollups will be returned. %s", e.getMessage()));
```

Comment:
Why not log the exception? By doing e.getMessage() we don't get the stack trace.